### PR TITLE
ENH / STY: improve dev package finder function, run pre-commit-hooks

### PR DESCRIPTION
**Pull request by tangkong**
## Description
A more robust dev package finder, and the style edits required to pass travis

## Motivation and Context
#332 

## How Has This Been Tested?
Tested on my own dev environment, as well as xpp's env.  
pkgutil seems to take only slightly longer than searching for `__init__.py` files

## Where Has This Been Documented?
This PR

image

### CODE REVIEW

File: hutch_python/env_version.py [hunk 1]
```diff
@@ -51,11 +57,12 @@ def get_standard_dev_pkgs() -> set[str]:
     pythonpath = os.environ.get('PYTHONPATH', '')
     if not pythonpath:
         return set()
-    pkg_names = set()
-    for part in pythonpath.split(':'):
-        if any(((s in part) for s in _dev_ignore_list)) or (not part):
-            continue
-        pkg_names.add(os.path.basename(os.path.dirname(part)))
+    paths = pythonpath.split(':')
```
Comment (klauer) [inline hutch_python/env_version.py:60]: I think `:` should be `os.pathsep`

File: hutch_python/env_version.py [hunk 2]
```diff
@@ -14,6 +15,11 @@
 _dev_ignore_list = ['ami', 'pdsapp']
 
 
+def not_ignored(path, ignores=_dev_ignore_list):
+    """ Return True if _dev_ignore_list isn't in path or empty"""
+    return not (any((s in path for s in ignores)) or not path)
```
Comment (klauer) [inline hutch_python/env_version.py:20]: Did I get this right?
`not (x or not y)` -> `not x and y` -> `y and not x`

```suggestion
    return bool(path and not any(s in path for s in ignores))
```

**Review by klauer — APPROVED** [review]
* pre-commit fixes look fine to me
* Couple code smell/things that could be improved (constructive criticism!)

File: hutch_python/env_version.py (hunk 2 above)
Comment (tangkong) [reply hutch_python/env_version.py:20]: logic is hard

**Comment by ZLLentz** [comment]
I like how `xpp` is naturally found as a dev package by this

File: hutch_python/env_version.py [hunk 3]
```diff
@@ -51,11 +57,12 @@ def get_standard_dev_pkgs() -> set[str]:
...
         return set()
-    pkg_names = set()
-    for part in pythonpath.split(':'):
-        if any(((s in part) for s in _dev_ignore_list)) or (not part):
-            continue
-        pkg_names.add(os.path.basename(os.path.dirname(part)))
+    paths = pythonpath.split(':')
+    valid_paths = filter(not_ignored, paths)
+    pkg_names = set(n.name for n in
+                    pkgutil.iter_modules(path=valid_paths)
```
Comment (ZLLentz) [inline hutch_python/env_version.py:65]: I think this was the best possible approach, leveraging the normal python package tools

**Review by ZLLentz — APPROVED** [review]
Runs fast and looks very correct

**Pull request merged**